lib/src/deriv_chart/interactive_layer/interactive_layer.dart

```diff
@@ -300,31 +413,214 @@ class _InteractiveLayerGestureHandlerState
     }
   }
 
+  // Update the interaction mode and notify listeners if needed
+  void _updateInteractionMode(InteractionMode mode) {
+    if (_currentInteractionMode != mode) {
+      setState(() {
+        _currentInteractionMode = mode;
+      });
+    }
+  }
+
+  // Method to cancel any active crosshair
+  void _cancelCrosshair() {
+    if (_currentInteractionMode == InteractionMode.crosshair) {
+      widget.crosshairController.onExit(const PointerExitEvent());
+      _updateInteractionMode(InteractionMode.none);
+    }
+  }
+
+  // Check if a point hits any drawing
+  bool _hitTestDrawings(Offset localPosition) {
+    for (final drawing
+        in widget.drawings.cast<InteractableDrawing<DrawingToolConfig>>()) {
+      if (drawing.hitTest(localPosition, epochToX, quoteToY)) {
+        return true;
+      }
+    }
```

Comment:
```suggestion
for (final drawing in widget.drawings) {
      if (drawing.hitTest(localPosition, epochToX, quoteToY)) {
        return true;
      }
    }
```

Can we just write this?